src/openrct2/interface/Window.cpp

```diff
@@ -916,7 +916,9 @@ void window_rotate_camera(rct_window* w, int32_t direction)
 
     // has something to do with checking if middle of the viewport is obstructed
     rct_viewport* other;
-    screen_get_map_xy(x, y, &x, &y, &other);
+    CoordsXY coords = screen_get_map_xy({ x, y }, &other);
+    x = coords.x;
```

Comment:
We should just use `coords` in all of the below calcs instead of reusing x and y